typescript/src/deposit-sweep.ts

```diff
@@ -420,3 +500,51 @@ export async function submitDepositSweepProof(
     vault
   )
 }
+
+/**
+ * Checks if a UTXO is owned by a provided key pair based on its previous output
+ * script.
+ * @dev The function assumes previous output script comes form the P2PKH or
+ *      P2WPKH UTXO.
+ * @param keyPair - A Signer object containing the public key and private key
+ *        pair.
+ * @param prevOutScript - A Buffer containing the previous output script of the
+ *        UTXO.
+ * @param network - The Bitcoin network configuration, i.e. mainnet or testnet.
+ * @returns A boolean indicating whether the derived address from the UTXO's
+ *          previous output script matches either of the P2PKH or P2WPKH
+ *          addresses derived from the provided key pair.
+ */
+export function ownsUtxo(
+  keyPair: Signer,
```

Comment:
Isn't it enough to use `publicKey:Hex`?

Reply:
Done in 6099e46a5c0bac81e7d1c9776f5d41712d1581d0.